src/iop/basecurve.c

```diff
@@ -948,25 +948,25 @@ static inline void apply_legacy_curve(
     const float *const table,
     const float *const unbounded_coeffs)
 {
+  const size_t npixels = (size_t)width * height;
 #ifdef _OPENMP
 #pragma omp parallel for default(none) \
-  dt_omp_firstprivate(height, width, in, out, mul, table, unbounded_coeffs) \
+  dt_omp_firstprivate(npixels) \
+  dt_omp_sharedconst(in, out, mul, table, unbounded_coeffs) \
   schedule(static)
 #endif
-  for(size_t k = 0; k < (size_t)width * height; k++)
+  for(size_t k = 0; k < 4*npixels; k += 4)
```

Comment:
this one may bite: it's (int) * (int)(size_t), so maybe instead of

```cpp
k < 4*npixels
```
do
```cpp
k < npixels*4
```

I hope LLVM 12 will warn about things liek that...

Reply:
(int)*(size_t) promotes the (int).   The reason (int)*(int)*(size_t) is problematic is that the first multiplication uses (int) and only its result gets promoted for the second.